server/services/party/party.service.ts

```diff
@@ -2,28 +2,31 @@ import {
     Injectable,
     ConflictException,
     InternalServerErrorException,
-}                          from '@nestjs/common'
-import {InjectRepository}  from '@nestjs/typeorm'
-import {RpcException}      from '@nestjs/microservices'
-import {Party}             from './entities/party'
-import {Repository}        from 'typeorm'
-import {PartyEmitter}      from './party.emitter'
-import {PartyLeaderOnline} from './actions'
+}                                           from '@nestjs/common'
+import {InjectRepository}                   from '@nestjs/typeorm'
+import {RpcException}                       from '@nestjs/microservices'
+import {Party}                              from './entities/party'
+import {Repository}                         from 'typeorm'
+import {PartyEmitter}                       from './party.emitter'
+import {MakePartyLeader, PartyLeaderOnline} from './actions'
+import {CharacterClient}                    from "../character/client/character.client";
 
 @Injectable()
 export class PartyService {
     constructor(
         private emitter: PartyEmitter,
         @InjectRepository(Party)
-        private repo: Repository<Party>
+        private repo: Repository<Party>,
+        private character: CharacterClient
```

Comment:
We should lookup characters against the character microservice to tell if someone is online